src/app/shared/object-collection/shared/listable-object/listable-object-component-loader.component.ts

```diff
@@ -104,14 +125,24 @@ export class ListableObjectComponentLoaderComponent implements OnInit {
       [
         [this.badges.nativeElement],
       ]);
-    (componentRef.instance as any).object = this.object;
+    (componentRef.instance as any).object = object;
     (componentRef.instance as any).index = this.index;
     (componentRef.instance as any).linkType = this.linkType;
     (componentRef.instance as any).listID = this.listID;
     (componentRef.instance as any).showLabel = this.showLabel;
     (componentRef.instance as any).context = this.context;
     (componentRef.instance as any).viewMode = this.viewMode;
     (componentRef.instance as any).value = this.value;
+
+    if ((componentRef.instance as any).reloadedObject) {
+      this.subs.push((componentRef.instance as any).reloadedObject.subscribe((reloadedObject: DSpaceObject) => {
+        if (reloadedObject) {
+          console.log('Reloaded Object from/to', this.object, reloadedObject);
+          this.object = reloadedObject;
+          this.instantiateComponent(reloadedObject);
+        }
+      }));
+    }
```

Comment:
I think the problem lies with this change.

Here, a component is created, and afterwards a subscription to an observable property of that component is created. When it emits, the component is recreated.

However the original component is never destroyed, nor is the subscription ever unsubscribed.

To make this work you should keep track of the componentref you create, and at the start of the method, check if it exists, and destroy it if it does. Same for the subscription: before you subscribe, check whether the subscription already exists and unsubscribe if it does.

However in general this seems like a bad idea. Rather than shoehorning in a way to reload here, I think you should set the list containing all the results to stale, which will cause it to be automatically reloaded. Any change we make to this component has a chance to break things in other places, as it is used for every dynamically loaded list.

Reply:
I'm 100% sure the old component is destroyed because I added logs to its onDestroy. Is destroyed because of the [viewContainerRef.clear();](https://github.com/DSpace/dspace-angular/blob/d015b6caefa06dfc72ce649e6f716cf1ea2c9095/src/app/shared/object-collection/shared/listable-object/listable-object-component-loader.component.ts#L119). Previous component is destroyed, emitter completed and subscription unsubscribed.

Reply:
@artlowel : I wonder if you might quickly summarize the alternative approach you mentioned in today's meeting?  

You noted that this sort of individual component reloading is a bad idea, and that ideally we should be finding a way to set this individual result to "stale" and perhaps reloading it via a service?  Did I understand that correctly, or am I misstating your idea?

As noted in today's meeting, @alemarte and @atarix83, it seems like we may need to move this ticket forward mostly "as-is", just to get the basic fix in place.  However, I want to better understand the "ideal" solution/implementation, so that we can create a follow-up task to replace this component reloading with a better solution.

Reply:
@tdonohue 
>You noted that this sort of individual component reloading is a bad idea, and that ideally we should be finding a way to set this individual result to "stale" and perhaps reloading it via a service? Did I understand that correctly, or am I misstating your idea?

I think replacing a single result that's part of a list is asking for problems yes, but the fact that the list was being reloaded is exactly the issue that's being reported in #721. I missed that yesterday. Because I was short on time, I only looked in to the specific error described in [@alemarte's comment](https://github.com/DSpace/dspace-angular/pull/978#issuecomment-775993429), not at the reason for the PR. I said we should set the list to stale and then it would automatically be re-retrieved, but that's likely exactly what happened before this PR and what's causing #721, so it wouldn't help in this case.

The main problem I see with the approach in this PR is that you have a list of objects coming from the my-dspace-page component. The objects get passed down to the child components. They pass them to their children etc. Halfway down the line this PRs swaps out one of those objects for a different one, and for all components below it, the new object is passed to the input, but all components above it don't know about that change and still contain references to the original object. I think that can lead to all sorts of unwanted side effects. If any component above the place the swap happens uses it for anything, you've got a confict. 

I think an ideal solution would be to change the list at the source: resultsRD$ in MyDSpacePage component. It should no longer be tied 1 to 1 to the result from the search service, but instead it should only be initialized with that result, and if we need to swap an object later, we swap it there. That would ensure all components in the chain are working with the same information, and it would fix the issues mentioned in #721

Reply:
@artlowel : I'm not sure I fully understand your "ideal solution" in the context of this PR.  Are you suggesting that we need to rethink the single object reload altogether & replace it with a "smarter" reload of the entire list (one that won't cause you to lose context within your list of results, like described in #721)?   

I'm just trying to determine proper next steps here...as from my tests yesterday & today, this PR currently isn't working well & I'm not yet sure how to move it forward.

Reply:
>Are you suggesting that we need to rethink the single object reload altogether & replace it with a "smarter" reload of the entire list (one that won't cause you to lose context within your list of results, like described in #721)?

Exactly

Reply:
@atarix83 and @alemarte: I'd appreciate your feedback here.  I fully realize this PR has been a lot of work so far, but I'm also finding that it's really not working as expected & reloading individual objects/components is proving to be quite difficult.  I'm also noticing that within the rest of the UI, we never reload individual objects in a list (even in similar features like the Admin Search, which has button actions as well. In that example, we instead use a popup modal when clicking the button to perform a confirmation...and then reload the list as needed).

Do we need to consider another solution here?  Perhaps reloading individual objects/components is a flawed approach altogether & we need to investigate a way to reload the list while keeping your context the same?  One example might be something like this:

* Suppose you are looking at Item 2 on page 3 of the list of PoolTasks
* You click "Claim" on Item 2.
* This reloads the *entire page/list*, but you are kept at the same location in the list. Ideally you "follow" the new ClaimedTask.
    * In most situations this should just mean the reloaded list _must come back in the same order_ & you remain on the same page of that results list
    * If somehow the list size has changed (e.g. other people claimed tasks at the same time, reducing the size of the list), then you should move to the page where the ClaimedTask was. (However, this also could be logged as a known bug for now as we determine a better way to solve it)

I'd appreciate your feedback here, as I'm honestly not sure how to proceed with this PR as-is.  It seems like it's becoming problematic to reload individual objects/components, and I'm not wanting to delay Beta 5 in order to design/build a complex solution to this problem.